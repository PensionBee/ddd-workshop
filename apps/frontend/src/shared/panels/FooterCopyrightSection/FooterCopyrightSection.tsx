import { GridContainer, GridItem, GridRow } from "@shared/components/Grid";
import { DisclaimerL } from "@shared/components/Typography";
import { Link } from "@shared/components/Link";

const currentYear = new Date().getFullYear();

export const FooterCopyrightSection: React.FC = (props) => {
  return (
    <GridContainer
      data-rc="FooterCopyrightSection"
      pt={{ xs: 20, lg: 50 }}
      pb={20}
      {...props}
    >
      <GridRow>
        <GridItem align="center">
          <DisclaimerL as="p" align="center">
            Â© Copyright {currentYear} PensionBee Ltd. Company registration:
            9354862. FCA Reference Number: 744931. Information
            Commissioner&apos;s Office registration: ZA131262
          </DisclaimerL>
          <DisclaimerL as="p" align="center">
            We use cookies to ensure that you get the best possible experience.
            By continuing to use our website you are agreeing to their use.{" "}
            <Link href="/privacy-policy#4-cookies">
              Find out more about cookies
            </Link>
            .
          </DisclaimerL>
        </GridItem>
      </GridRow>
    </GridContainer>
  );
};
