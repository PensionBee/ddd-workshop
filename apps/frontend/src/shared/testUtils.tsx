import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
