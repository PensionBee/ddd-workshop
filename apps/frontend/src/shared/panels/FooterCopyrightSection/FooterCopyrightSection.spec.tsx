import { render } from "@shared/testUtils";
import { FooterCopyrightSection } from "./FooterCopyrightSection";

describe("FooterCopyrightSection", () => {
  const TEST_ID = "FooterCopyrightSection";
  it("should render", () => {
    const { getByTestId, asFragment } = render(
      <FooterCopyrightSection data-testid={TEST_ID} />
    );
    expect(getByTestId(TEST_ID)).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();
  });
});
