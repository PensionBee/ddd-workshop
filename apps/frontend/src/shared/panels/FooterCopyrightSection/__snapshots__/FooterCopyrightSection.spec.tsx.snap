// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterCopyrightSection > should render 1`] = `[Function]`;

exports[`FooterCopyrightSection should render 1`] = `[Function]`;
